Subject: node_device_conf: Don't leak @physical_function in virNodeDeviceGetPCISRIOVCaps

The pci_dev->physical_function is rewritten in
virPCIGetPhysicalFunction() to a newly allocated pointer.
Therefore, we must free the old one to avoid memleak.

Author: Jiang Kun <jiang.kun2@zte.com.cn>
Origin: upstream, https://libvirt.org/git/?p=libvirt.git;a=commit;h=38816336
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1844455
Last-Update: 2020-02-20

diff --git a/src/conf/node_device_conf.c b/src/conf/node_device_conf.c
index e51371de89e3..618ce8e00e5c 100644
--- a/src/conf/node_device_conf.c
+++ b/src/conf/node_device_conf.c
@@ -2509,6 +2509,7 @@ virNodeDeviceGetPCISRIOVCaps(const char *sysfsPath,
     for (i = 0; i < pci_dev->num_virtual_functions; i++)
        VIR_FREE(pci_dev->virtual_functions[i]);
     VIR_FREE(pci_dev->virtual_functions);
+    VIR_FREE(pci_dev->physical_function);
     pci_dev->num_virtual_functions = 0;
     pci_dev->max_virtual_functions = 0;
     pci_dev->flags &= ~VIR_NODE_DEV_CAP_FLAG_PCI_VIRTUAL_FUNCTION;
-- 
2.24.1

